src/word.rs

```diff
@@ -0,0 +1,100 @@
+use super::{Felt, StarkField};
+use crate::hash::rpo::RpoDigest;
+use core::ops::Deref;
+
+// WORD
+// ================================================================================================
+
+/// A group of four field elements in the Miden base field.
+pub type Word = [Felt; WORD_SIZE];
+
+// CONSTANTS
+// ================================================================================================
+
+/// Number of field elements in a word.
+pub const WORD_SIZE: usize = 4;
+
+// CANONICAL WORD
+// ================================================================================================
+
+/// A `[Word]` in canonical representation.
+///
+/// `[Felt]` won't implement `[PartialOrd]` because its inner number cannot be ordered as it is
+/// often in Montgomery form. If it was to implement `PartialOrd`, it would have to first fetch a
+/// canonical `u64`, and then call its implementation of `PartialOrd`. However, this is not only
+/// inefficient, but also would lead to several canonical conversions, strictly unnecessary, being
+/// performed during simple sort operations. One data structure that would be severely affected by
+/// that is `BTreeMap`, that we heavily use, and depends on `Ord` for its keys.
+///
+/// This structure allows safe and correct ordering of words.
+#[derive(Copy, Clone, Debug, Default, PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub struct CanonicalWord([u64; WORD_SIZE]);
```

Comment:
I would remove this struct from this PR. I'm not yet sure we need it, and if we do, I think we can add it in the SMT PR.

Reply:
Done!